backend/fms_core/viewsets/full_sample.py

```diff
@@ -0,0 +1,216 @@
+
+        serializer = SampleSerializer(sample)
+        data = serializer.data
+        return Response(data)
+
+    def update(self, request, *args, **kwargs):
+        full_sample = request.data
+
+        #Retrive the sample to update
+        try:
+            sample_to_update = Sample.objects.get(pk=full_sample['id'])
+        except Exception as e:
+            raise ValidationError(e)
+
+        #Update individual fields
+        if full_sample['name']:
+            sample_to_update.name = full_sample['name']
+        if full_sample['container']:
+            try:
+                new_container = Container.objects.get(pk=full_sample['container'])
+            except Exception as e:
+                raise ValidationError(e)
+            sample_to_update.container = new_container
+        if full_sample['coordinates']:
+            sample_to_update.coordinates = full_sample['coordinates']
+        if full_sample['volume']:
+            sample_to_update.volume = full_sample['volume']
+        if full_sample['concentration']:
+            sample_to_update.concentration = float_to_decimal(full_sample['concentration'])
+        if full_sample['depleted'] is not None:
+            sample_to_update.depleted = full_sample['depleted']
+
+        #Save the new sample
+        try:
+            sample_to_update.save()
+        except ValidationError as err:
+            raise ValidationError(err)
+
+        #Return new sample
+        serializer = SampleSerializer(sample_to_update)
+        data = serializer.data
+        return Response(data)
+
+
+
+    @action(detail=False, methods=["get"])
+    def list_export(self, _request):
+        serializer = FullSampleExportSerializer(self.filter_queryset(self.get_queryset()), many=True)
+        return Response(serializer.data)
+
+    @action(detail=False, methods=["get"])
+    def summary(self, _request):
+        """
+        Returns summary statistics about the current set of samples in the
+        database.
+        """
+
+        experimental_groups = Counter()
+        for eg in DerivedSample.objects.values_list("experimental_group", flat=True):
+            experimental_groups.update(eg)
+
+        return Response({
+            "total_count": Sample.objects.all().count(),
+            "kinds_counts": {
+                c["sample_kind"]: c["sample_kind__count"]
+                for c in DerivedSample.objects.values("sample_kind").annotate(Count("sample_kind"))
+            },
+            "tissue_source_counts": {
+                c["tissue_source"]: c["tissue_source__count"]
+                for c in DerivedSample.objects.values("tissue_source").annotate(Count("tissue_source"))
+            },
+            "collection_site_counts": {
+                c["collection_site"]: c["collection_site__count"]
+                for c in Biosample.objects.values("collection_site").annotate(Count("collection_site"))
+            },
+            "experimental_group_counts": dict(experimental_groups),
```

Comment:
Are these counts used somewhere ? we may want to get rid of some of them if they are not useful. summary is called very often and having those extra counts (the total is displayed in the dashboard) may be wasted.

Reply:
I don't think so, imo we can get rid of them indeed